noxfile.py

```diff
@@ -37,6 +37,8 @@ def set_environment_variables(env_dict, session):
     for key, value in env_dict.items():
         session.env[key] = value
 
+    session.env["PYTHONIOENCODING"] = "utf-8"
+
```

Comment:
Just looked at this change, could you please move it to the `PYBAMM_ENV` dictionary at the top of the file? It will then be applied to all sessions as it is needed.